src/main/java/net/snowflake/client/core/FileCacheManager.java

```diff
@@ -153,47 +152,46 @@ FileCacheManager build()
       if (Constants.getOS() == Constants.OS.WINDOWS)
       {
         this.cacheDir = new File(
-            new File(new File(new File(homeDir,
-                                       "AppData"), "Local"), "Snowflake"), "Caches");
+            new File(new File(new File(homeDir, "AppData"), "Local"), "Snowflake"), "Caches");
       }
       else if (Constants.getOS() == Constants.OS.MAC)
       {
-        this.cacheDir = new File(new File(new File(homeDir,
-                                                   "Library"), "Caches"), "Snowflake");
+        this.cacheDir = new File(new File(new File(homeDir, "Library"), "Caches"), "Snowflake");
       }
       else
       {
         this.cacheDir = new File(new File(homeDir, ".cache"), "snowflake");
       }
     }
-    if (!this.cacheDir.exists() && !this.cacheDir.mkdirs())
+
+    if (!this.cacheDir.mkdirs() && !this.cacheDir.exists())
     {
-      throw new RuntimeException(
-          String.format(
-              "Failed to locate or create the cache directory: %s", this.cacheDir)
-      );
+      LOGGER.debug("Cannot create the cache directory {}. Giving up.", this.cacheDir.getAbsolutePath());
+      return this;
```

Comment:
No RuntimeException.
If no cache directory exists, it will not use a cache file.